site/app/controllers/HomePageController.php

```diff
@@ -290,10 +290,40 @@ public function createCoursePage() {
                 'showCourseCreationPage',
                 $faculty ?? null,
                 $this->core->getUser()->getId(),
-                $this->core->getQueries()->getAllUnarchivedSemester(),
+                //$this->core->getQueries()->getAllUnarchivedSemester(),
+                $this->core->getQueries()->getAllTerms(),
                 $this->core->getUser()->getAccessLevel() === User::LEVEL_SUPERUSER,
                 $this->core->getCsrfToken()
             )
         );
     }
+
+    /**
+     * @Route("/term/new", methods={"POST"})
+     * @return Response
+     */
+    public function addNewTerm() {
+        if (isset($_POST['term_id']) && isset($_POST['term_name']) && isset($_POST['start_date']) && isset($_POST['end_date'])) {
+            $term_id = $_POST['term_id'];
+            $term_name = $_POST['term_name'];
+            $start_date = $_POST['start_date'];
+            $end_date = $_POST['end_date'];
+
+            $terms = $this->core->getQueries()->getAllTerms();
+            if (in_array($term_id, $terms)) {
+                $this->core->addErrorMessage("Term id already exists.");
+            }
+            else if ($end_date < $start_date) {
```

Comment:
I think this is using normal string comparison instead of comparing date objects. String comparison might work I am not 100% sure so I thought I would bring it up.

Reply:
i think it doesn't matter if they are string compared as long as they have the same format, they are submitted to the database as date formats as well.